db/query_test.go

```diff
@@ -568,3 +568,135 @@ func countQueryResults(results sgbucket.QueryResultIterator) int {
 	}
 	return count
 }
+
+func TestQueryChannelsActiveOnlyWithLimit(t *testing.T) {
+
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("This test require Couchbase Server")
+	}
+
+	db, testBucket := setupTestDB(t)
+	defer testBucket.Close()
+	defer tearDownTestDB(t, db)
+
+	// Create doc1, revision 1-a
+	body := Body{"channels": []string{"ABC"}, "name": "Alice"}
+	doc, _, err := db.PutExistingRevWithBody("doc1", body, []string{"1-a"}, false)
+	require.NoError(t, err, "Couldn't create document")
+	startSeq := doc.Sequence
+
+	// Tombstone doc1, revision 2-a
+	body[BodyDeleted] = true
+	doc, _, err = db.PutExistingRevWithBody("doc1", body, []string{"2-a", "1-a"}, false)
+	require.NoError(t, err, "Couldn't create document")
+
+	// Create doc2 and two conflicting changes:
+	body["name"] = "Bob"
+	_, _, err = db.PutExistingRevWithBody("doc2", body, []string{"1-a"}, false)
+	require.NoError(t, err, "Couldn't create document")
+
+	body["name"] = "Owen"
+	_, _, err = db.PutExistingRevWithBody("doc2", body, []string{"2-b", "1-a"}, false)
+	require.NoError(t, err, "Couldn't create revision 2-b of doc2")
+
+	body["name"] = "Chloe"
+	_, _, err = db.PutExistingRevWithBody("doc2", body, []string{"2-a", "1-a"}, false)
+	require.NoError(t, err, "Couldn't create revision 2-a of doc2")
+
+	// Tombstone the non-winning branch of a conflicted document
+	body[BodyDeleted] = true
+	_, _, err = db.PutExistingRevWithBody("doc2", body, []string{"3-a", "2-a"}, false)
+	assert.NoError(t, err, "Add 3-a (tombstone)")
+
+	// Create doc3
+	body["name"] = "Emily"
+	doc, _, err = db.PutExistingRevWithBody("doc3", body, []string{"1-a"}, false)
+	require.NoError(t, err, "Couldn't create revision 1-a of doc3")
+	endSeq := doc.Sequence
+
+	// Get changes in ABC channel with limit and activeOnly true
+	entries, err := db.getChangesInChannelFromQuery("ABC", startSeq, endSeq, 2, true)
```

Comment:
Since there are only two active docs, testing with limit 2 here isn't evaluating whether limit is working.  The test would generally be more robust if we're writing quite a few more docs (i.e. 10 of 'doc1', 10 of 'doc2', etc), and then have the limit be something much smaller than the expected resultset.

Reply:
Make sense. Added few more docs to check the limit functionality.